src/J2N/Collections/Generic/EnumerableHelpers.cs

```diff
@@ -19,7 +19,7 @@ internal static partial class EnumerableHelpers
     {
         /// <summary>Calls Reset on an enumerator instance.</summary>
         /// <remarks>Enables Reset to be called without boxing on a struct enumerator that lacks a public Reset.</remarks>
-        internal static void Reset<T>(ref T enumerator) where T : IEnumerator => enumerator.Reset();
+        internal static void Reset<T>(in T enumerator) where T : IEnumerator => enumerator.Reset();
```

Comment:
This was `ref` upstream. Let's revert this. Most enumerators are structs, and removing the ref will cause the instance to be copied.

https://github.com/dotnet/runtime/blob/v10.0.0-rc.2.25502.107/src/libraries/Common/src/System/Collections/Generic/EnumerableHelpers.cs#L15